tests/test_populate_catalog.py

```diff
@@ -652,3 +655,26 @@ def should_add_link_to_root_catalog_in_series(
                 expected_root_catalog_links
                 == load(smart_open.open(root_url, mode="rb"))[STAC_LINKS_KEY]
             )
+
+
+@mark.infrastructure
+@patch("geostore.populate_catalog.task.read_file")
+def should_log_error_message_when_an_exception_is_raised_trying_to_update_catalog(
+    pystac_read_file_mock: MagicMock,
+) -> None:
+
+    error = Exception()
```

Comment:
There's an `any_exception_class` function you might want to use here. It does a couple useful things:

1. Ties the mock side effect explicitly to the `raises` clause, since you have to reference the same class in both places for the test to pass.
2. `Exception` is the superclass of *all* standard error classes, so I believe `raises(Exception)` will pass no matter which exception is raised. By using a *specific* and *unique* exception we ensure that any other raised exceptions will cause the test to fail.

(We should probably change `any_exception_class` to check that the class name isn't defined yet before creating the class, just to make sure we don't accidentally override an existing exception class during the test.)

Reply:
Excellent. Was not aware of the `any_exception_class` function. Thanks for logging #2093. Would that be a prerequisite for this PR to be approved?

Reply:
Nah, that's definitely a separate PR.